### PR TITLE
[bitnami/vault] feat: :tada: Add chart

### PR DESCRIPTION
**Pull request by javsalgar**
### Description of the change

This PR adds a chart for [HashiCorp Vault](https://www.vaultproject.io/) based on the upstream  https://github.com/hashicorp/vault-helm chart. We mimicked the same behavior (it requires a manual unsealing to make it work). 

### Benefits

New item in the catalog.

### Possible drawbacks

n/a

### Applicable issues

- fixes #

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/vault/values.yaml [hunk 1]
```diff
@@ -0,0 +1,1581 @@
...
+    ## @param server.serviceAccount.automountServiceAccountToken Automount service account token for the server service account
+    ##
+    automountServiceAccountToken: true
+
+  ## @section Source Conttroller Persistence Parameters
+  ## ref: https://kubernetes.io/docs/user-guide/persistent-volumes/
+  ##
+  persistence:
+    ## @param server.persistence.enabled Enable persistence using Persistent Volume Claims
+    ## (NOTE: Disabled by default in upstream flux configuration)
```
Comment (fmulero) [inline bitnami/vault/values.yaml:617]: I think this note doesn't apply in this case

File: bitnami/vault/values.yaml [hunk 2]
```diff
@@ -0,0 +1,1581 @@
...
+
+## @section Vault CSI Provider Parameters
+##
+csiProvider:
+  ## @param csiProvider.enabled Enable Vault CSI Provider
+  ## (NOTE: This is disabled in upstream Vault chart because it requires plugins in the K8s cluster)
+  ##
+  enabled: false
+  ## Bitnami Vault CSI Provider image
+  ## ref: https://hub.docker.com/r/bitnami/fluxcd-csi-provider/tags/
```
Comment (fmulero) [inline bitnami/vault/values.yaml:721]: Flux?? I think this is a typo

**Review by fmulero — CHANGES_REQUESTED** [review]
Great work @javsalgar!!!

I left some comments, main of them are about the labels and selectors. Could you review them?

File: .vib/vault/cypress/cypress/fixtures/operations.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
```
[the remark this answers is not in the document]
Comment (javsalgar) [reply .vib/vault/cypress/cypress/fixtures/operations.json:1]: FIxed

File: bitnami/vault/Chart.lock [hunk 4]
```diff
@@ -0,0 +1,6 @@
+dependencies:
```
Comment (fmulero) [inline bitnami/vault/Chart.lock:1]: This file has to be generated again to build the chart properly

Comment (javsalgar) [reply bitnami/vault/Chart.lock:1]: Fixed

File: bitnami/vault/templates/injector/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+      {{- end }}
+      {{- if .Values.injector.podSecurityContext.enabled }}
+      securityContext: {{- omit .Values.injector.podSecurityContext "enabled" | toYaml | nindent 8 }}
+      {{- end }}
+      {{- if .Values.injector.terminationGracePeriodSeconds }}
+      terminationGracePeriodSeconds: {{ .Values.injector.terminationGracePeriodSeconds }}
+      {{- end }}
+      {{- if .Values.injector.initContainers }}
+      initContainers:
+        {{- include "common.tplvalues.render" (dict "value" .Values.initContainers "context" $) | nindent 8 }}
```
Comment (migruiz4) [inline bitnami/vault/templates/injector/deployment.yaml:78]: There is a typo here:
```suggestion
        {{- include "common.tplvalues.render" (dict "value" .Values.injector.initContainers "context" $) | nindent 8 }}
```

File: bitnami/vault/templates/server/statefulset.yaml [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+            - |
+              mkdir -p {{ .Values.persistence.mountPath }}
+              find {{ .Values.persistence.mountPath }} -mindepth 1 -maxdepth 1 -not -name ".snapshot" -not -name "lost+found" | xargs --no-run-if-empty chown -R {{ .Values.server.containerSecurityContext.runAsUser }}:{{ .Values.server.podSecurityContext.fsGroup }}
+          {{- if .Values.volumePermissions.containerSecurityContext.enabled }}
+          securityContext: {{- omit .Values.volumePermissions.containerSecurityContext "enabled" | toYaml | nindent 12 }}
+          {{- end }}
+          resources: {{- toYaml .Values.volumePermissions.resources | nindent 12 }}
+          volumeMounts:
+            - name: data
+              mountPath: {{ .Values.persistence.mountPath }}
```
Comment (migruiz4) [inline bitnami/vault/templates/server/statefulset.yaml:93]: The value `persistence.mountPath` should be `server.persistence.mountPath`
```suggestion
              mountPath: {{ .Values.server.persistence.mountPath }}
```

File: bitnami/vault/templates/server/statefulset.yaml [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+        {{- if and .Values.volumePermissions.enabled .Values.server.containerSecurityContext.enabled }}
+        - name: volume-permissions
+          image: {{ include "vault.volumePermissions.image" . }}
+          imagePullPolicy: {{ .Values.volumePermissions.image.pullPolicy | quote }}
+          command:
+            - sh
+            - -c
+            - |
+              mkdir -p {{ .Values.persistence.mountPath }}
+              find {{ .Values.persistence.mountPath }} -mindepth 1 -maxdepth 1 -not -name ".snapshot" -not -name "lost+found" | xargs --no-run-if-empty chown -R {{ .Values.server.containerSecurityContext.runAsUser }}:{{ .Values.server.podSecurityContext.fsGroup }}
```
Comment (migruiz4) [inline bitnami/vault/templates/server/statefulset.yaml:86]:
```suggestion
              mkdir -p {{ .Values.server.persistence.mountPath }}
              find {{ .Values.server.persistence.mountPath }} -mindepth 1 -maxdepth 1 -not -name ".snapshot" -not -name "lost+found" | xargs --no-run-if-empty chown -R {{ .Values.server.containerSecurityContext.runAsUser }}:{{ .Values.server.podSecurityContext.fsGroup }}
```

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Please take a look at my comments

**Review by migruiz4 — DISMISSED** [review]
LGTM!

**Pull request merged**